src/Microsoft.Azure.WebJobs.Host/Listeners/FunctionListenerDecorator.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Host.Listeners
+{
+    internal class FunctionListenerDecorator : IListenerDecorator
```

Comment:
Previously this "decorator" and Singleton were both listener wrappers that were hardcoded in the listener pipeline. Refactoring these out into decorators.